File: tests/llvm/hardlift/benchmark/python/linear_dodge_8.py
from typing import List

def linear_dodge_8(base: List[List[int]], active: List[List[int]]) -> List[List[int]]:
  output: List[List[int]] = []
  m = len(base)
  n = len(base[0])
  for i in range(m):
    curr_row: List[int] = []
    for j in range(n):
      pixel = base[i][j] + active[i][j]
      curr_row.append(pixel)
    output.append(curr_row)
  return output

File: tests/llvm/hardlift/benchmark/cpp/linear_dodge_8.cc
#include <iostream>
#include <vector>
#include <chrono>

#include "utils.h"

using namespace std;
using namespace std::chrono;

vector<vector<float>> linear_dodge_8(vector<vector<float>> base, vector<vector<float>> active) {
    vector<vector<float>> out;
    int m = base.size();
    int n = base[0].size();
    for (int row = 0; row < m; row++) {
        vector<float> row_vec;
        for (int col = 0; col < n; col++) {
            float pixel = base[row][col] + active[row][col];
            row_vec.push_back(pixel);
        }
        out.push_back(row_vec);
    }
    return out;
}


int main() {
    mat_timer(&linear_dodge_8);
    return 0;
}
